### DISCUSSION TITLE
cluster_events balancer:target Update suggestion

### DISCUSSION THREAD

**Discussion by tianjunqi** [Kong/kong#7348] [Ideas and feature requests]
### Summary
Every time the balancer is updated and rebuilt, the service performance overhead is too large, which affects the stability of the request。
It is recommended to process only the changed target address each time。

### to worker_events node handler
Event data increase `target.id`
`fmt("%s:%s:%s", operation, target.upstream.id, target.id)`

### cluster_events handler
Use the following method to process the target address：
```
assert(balancer:addHost(name, port, target.weight))
assert(balancer:removeHost(name, port))
```

**Comment by locao** [comment]
Hi @tianjunqi! Thank you for your suggestion. We are already making a large rewrite on this code and this is one of our concerns. We'll let you know when this is released.
